src/numericalAnalysis.cpp

```diff
@@ -0,0 +1,119 @@
+#include "mantella_bits/numericalAnalysis.hpp"
+
+// C++ standard library
+#include <cmath>
+#include <limits>
+
+// Mantella
+#include "mantella_bits/assert.hpp"
+
+namespace mant {
+  double brent(
+      std::function<double(double)> objectiveFunction,
+      double lowerBound,
+      double upperBound,
+      double xTolerance) {
+    double lowerBoundObjectiveValue = objectiveFunction(lowerBound);
+    double upperBoundObjectiveValue = objectiveFunction(upperBound);
+
+    verify(lowerBoundObjectiveValue * upperBoundObjectiveValue <= 0.0, "brent: The lower bound and upper bound objective value must have a different sign.");
+
+    double previousUpperBound = lowerBound; // only on initialisation: previousUpperBound = lowerBound
+    double previousUpperBoundObjectiveValue = lowerBoundObjectiveValue;
+    double boundDistance = upperBound - lowerBound;
+    double toleranceCriteria = boundDistance;
+
+    double middleOfBounds;
+    double boundObjectiveValueRatio;
+    double interpolationNumerator;
+    double interpolationFactor;
+    double boundPreviousBoundRatio;
+
+    int iter = 0;
+    int maxIter = 1000;
```

Comment:
Please avoid any form of abbreviation if possible and use `maximalNumberOfIterations` instead. `maximalIteration` alone is also problematic, as it could also be interpret as an iteration pointer (like `.cend`) instead of a number.